truffle/src/main/ruby/core/proc.rb

```diff
@@ -33,18 +41,31 @@ class Proc
 
   def curry(curried_arity = nil)
     if lambda? && curried_arity
+
       if arity >= 0 && curried_arity != arity
-        raise ArgumentError, "Wrong number of arguments (%i for %i)" % [
-          curried_arity,
-          arity
-        ]
+        raise ArgumentError, "Wrong number of arguments (#{curried_arity} for #{arity})"
       end
 
-      if arity < 0 && curried_arity < (-arity - 1)
-        raise ArgumentError, "Wrong number of arguments (%i for %i)" % [
-          curried_arity,
-          -arity - 1
-        ]
+      if arity < 0
+        total_args = 0
+        req_args = 0
```

Comment:
I was torn on this, required and total feels like it expresses the parameter semantics while min and max express our use of them here. In the end I switched to min and max because it makes the code line up nicely. :smile: